libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -45,3 +50,114 @@ void AP_BattMonitor_Backend::Log_Write_BCL(const uint8_t instance, const uint64_
     static_assert(ARRAY_SIZE(_state.cell_voltages.cells) == ARRAY_SIZE(cell_pkt.cell_voltages),
                     "Battery cell number doesn't match in library and log structure");
 }
+
+void AP_BattMonitor_Backend::Log_Write_BATI(const uint8_t instance)
+{
+    exclusive_send_or_log_smart_battery_info(instance, true);
+}
+
+bool AP_BattMonitor_Backend::send_mavlink_smart_battery_info(const uint8_t instance, const mavlink_channel_t chan)
+{
+    return exclusive_send_or_log_smart_battery_info(instance, false, chan);
+}
+
+// Exclusively either send or log the smart battery info. If log_flag true logs BATI message if false sends mavlink message
+bool AP_BattMonitor_Backend::exclusive_send_or_log_smart_battery_info(const uint8_t instance, bool log_flag, const mavlink_channel_t chan)
```

Comment:
we don't normally have a single method that both sends status via mavlink and log to onboard logs. see "exclusive_send_or_log_smart_battery_info"

Reply:
@rmackay9 
Yeah this was my way to save flash space. I think it was 273 bytes in the code as it is currently. 

But it adds complexity doing it like this. I'm guessing clearer code is preferable?